rfd/0136-modern-signature-algorithms.md

```diff
@@ -0,0 +1,777 @@
+---
+authors: Nic Klaassen (nic@goteleport.com)
+state: draft
+---
+
+# RFD 136 - Modern Signature Algorithms
+
+## Required Approvers
+
+* Engineering: (@jakule || @espadolini)
+* Security: (@reedloden || @jentfoo) && Doyensec
+* Product: @klizhentas
+
+## What
+
+This RFD proposes the adoption of a new set of asymmetric key types and
+signature algorithms across all Teleport protocols that surpass the 2048-bit RSA
+keys currently used in terms of security and performance.
+
+## Why
+
+Currently Teleport uses 2048-bit RSA keypairs for all internal Certificate
+Authorities and all user and host keys.
+
+More modern algorithms based on Elliptic Curve Cryptography offer better
+security properties with smaller keys that offer better performance for keypair
+generation, signing operations, and signature verification.  Some of the more
+restrictive security policies are starting to reject RSA2048 (e.g.
+[RHEL 8's FUTURE policy](https://access.redhat.com/articles/3642912)).
+
+## Details
+
+### Summary
+
+We will introduce the concept of signature algorithm "suites" that will
+configure the key types and signature algorithms used across Teleport.
+This will be a cluster-wide configuration available in both the `teleport.yaml`
+config of Teleport Auth servers (so that initial CA keys generated at first
+startup can be configured) and in the `cluster_auth_preference` (so that Cloud
+users can optionally change the suite used in their cluster).
+
+The initial set of suites we will introduce is:
+
+* `legacy`
```

Comment:
Instead of `legacy` should we call this `rsa-v1` or `speedy-v1`?

Reply:
I wanted to call it `legacy` to suggest that you should prefer not to use it, and sasha used that name in [this](https://github.com/gravitational/teleport/pull/28110#issuecomment-1622432407) comment. `speedy-v1` doesn't seem right, RSA is much slower for key generation (which we do a decent amount of, even server side). From what I've read it also seems a bit slower for signing ops, but faster for verification - anyway, I think `speedy-v1` makes it sound "good" which we probably want to avoid. I wouldn't be against calling it `rsa-v1`, though

Reply:
I agree `speedy` is probably not accurate, but still think `rsa-v1` may be a good naming.  I will defer to your preference

Reply:
let's call it `legacy` as we want people to migrate off it